Yandex_Sprint_7

lexus322 opened the pull request.

lexus322 committed:
Add test + allure results

lexus322 committed:
Add pox.mxl

mkouznetsov-git commented on src/test/java/test/CreateOrderColorFieldTest.java:
```diff
@@ -0,0 +1,52 @@
+@DisplayName("Создание заказа с параметаризиронными данными для цвета")
+@RunWith(Parameterized.class)
+public class CreateOrderColorFieldTest extends BaseOrderTest {
+    private final String testName;
```
Переменная присваивается, но не используется.

mkouznetsov-git replied:
Не исправлено.

mkouznetsov-git replied:
Третий раз обращаю внимание на это место. Надо либо задействовать и использовать эту переменную, либо удалить.

lexus322 replied:
@mkouznetsov-git 
Переменная testname используется для получения названия поля и вынесения его в название теста
Я вам писал в последнем коммите что она используется для названия теста

lexus322 replied:
В послденем коммите удалил перменную

lexus322 replied:
Я через нее доставал имя enuma что передать ее как параметр для вывода в тесте

mkouznetsov-git replied:
Ок, я понял.

mkouznetsov-git commented on src/test/java/test/CreateCourierTest.java:
```diff
@@ -0,0 +1,98 @@
+        Response response = courierAction.postRequestCreateCourier(courierCard);
+        response.then().assertThat().body("ok", Matchers.equalTo(true))
+                .and().statusCode(SC_CREATED);
+        deleteTestCourier();
```
Если тест упадёт где-то раньше, то до удаления дело не дойдёт. Надо перенести удаление в метод с аннотацией After. Это надо сделать для всех случаев в этом классе и в классе LogInCourierTest.

lexus322 replied:
@mkouznetsov-git в связи с тем что у меня в половине случаев при создание курьера, само создание не происходит(негативные сценарии, такие как создание курьера без логина или без пароля) я не вызываю метод удаление данных, возникает вопрос как мне прописать афтер для разных тестов внутри метода
Нашел два вариант
1. Простой вынести тесты где создание происходит в отдельный класс и там уже прописать ему афтер с удалением тестовых данных
2. Использовать TestNG там есть аннотация @AfterMethod и я смогу контролировать где вызывать афтер,но это вличет переписыванием большого количества кода

mkouznetsov-git replied:
Тут не сложно. Надо просто проверять токен. Если пустой, значит курьер не создался, можно не удалять.
Рассчитывать на то, что у тебя негативные тесты, и они сто процентов будут выдавать то, что ты ожидаешь -- это неверно. Потому что если будет какая-то ошибка в системе, и курьер всё-таки создастся, это будет баг. А ты при этом не удалишь тестовые записи созданных курьеров.

lexus322 replied:
Добавил аннотацию, чтобы в вариантах где не создается курьер поле всегда было и можно было проверить что оно пустое
Также добавил для собственной проверки вывод в консоль о том что удаление произошло или оно не требуется

lexus322 committed:
Исправления от 23.05
Негативные и позитивные сценарии разделенны для добавления аннотации @After для удаленния созданных данных

lexus322 committed:
Исправления от 25.05
Добавлена проверка по id для удаления тестовых данных внутри анотации @after
Переменная testname используется для получения названия поля и вынесения его в название теста
Мы нигде не получаем токен, поэтому решил сделать проверку по id, считаю что это приемлемо так как при успешном создание id есть при неспешном id отсуствует, поэтому мы можем понять где удалять данные нужно а где нет

lexus322 committed:
Исправления от 26.05
Удалена перменная testname
